src/lib.rs

```diff
@@ -186,6 +189,55 @@ pub fn find_configuration_file() -> String {
     }
 }
 
+pub fn encrypt_single_file(
+    input_file: &str,
+    output_file: &str,
+    encryption_key_string: Option<String>,
+) {
+    let encryption_key = match encryption_key_string {
+        Some(encryption_key_string) => match EncryptionKey::from_str(&encryption_key_string) {
+            Ok(encryption_key) => encryption_key,
+            Err(err) => {
+                println!("{:?}", err);
+                std::process::exit(err as i32);
+            }
+        },
+        None => crate::encryption::generate_key(),
```

Comment:
This part is a little interesting to me. If we generate a key ourselves and not tell the user what the key is how can they decrypt it after? The answer to that question will change how I review some of the code around it, so I am going to hold off on commenting on those for now.

Reply:
This was also addressed in d23064f8f18e6f36c85f1bd2b7209a23060e651a, because yeah, that made no sense 🙈